common/flatpak-dir.c

```diff
@@ -2092,10 +2092,12 @@ flatpak_dir_update_appstream (FlatpakDir          *self,
           if (child_repo == NULL)
             return FALSE;
 
-          if (!flatpak_dir_remote_fetch_summary (self, remote,
+#ifndef FLATPAK_ENABLE_P2P
```

Comment:
I wonder if not fetching the summary just because we have p2p enabled is the right approach.
Why not check the error instead? So when there's a connection available it does fetch the summary, and when not, it continues.

Reply:
Also, as much as I usually try to avoid comments in the code, I think this is a case where we could use a comment explaining either why we're skipping the block or checking the error.

Reply:
It seems unnecessary to do it anyway and check the error. It's going to be deleted at the end of the function anyway.

Reply:
I pushed a fixup to add comments, and a fixup to fetch the summary when updating appstream if there's a collection ID set.